derive-tag/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn;
+
+#[proc_macro_derive(ToTag)]
+pub fn to_tag_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_to_tag(&ast)
+}
+
+fn impl_to_tag(ast: &syn::DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+
+    let gen = quote! {
+        impl ToTag for #name {
+            fn to_tag(&self) -> u8 {
+                unsafe { *(self as *const Self as *const u8) }
+            }
+        }
+    };
+
+    gen.into()
+}
+
+#[proc_macro_derive(FromTag)]
+pub fn from_tag_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_from_tag(&ast)
+}
+
+fn impl_from_tag(ast: &syn::DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+
+    let mut match_variants = quote!();
+
+    match &ast.data {
+        syn::Data::Enum(syn::DataEnum {
+            variants,
+            enum_token: _x,
+            brace_token: _y,
```

Comment:
Minor improvement, or it's just style, you can use `..` instead of naming all of the other properties.

Maybe it's better to keep the way you've done so it's easier to know what's inside in the future 🤷 , you decide 😉